MekHQ/src/mekhq/campaign/market/ContractMarket.java

```diff
@@ -69,6 +69,11 @@
     public final static int CLAUSE_TRANSPORT = 3;
     public final static int CLAUSE_NUM = 4;
 
+    /**
+     * An arbitrary maximum number of attempts to generate a contract.
+     */
+    private final static int MAXIMUM_GENERATION_RETRIES = 3;
```

Comment:
We should probably make this a campaign option.

Reply:
I'm working on a follow-up with unit tests, I'll expand that there.

Reply:
Not really sure why. From a user perspective, there's no clear relationship between it and program behavior.

Reply:
I guess to help with odd issues.